### ISSUE TITLE
Output ODEs for covariance of LNA?

### ISSUE THREAD

**Issue by sdwfrost** [fintzij/stemr#1]
I've looked through the code and help pages, but I can't find a simple way to be able to simulate the means and covariances of the LNA over time - the simulation just returns draws from the LNA (at least under the options in the SIR vignette). Any guidance on how to do this?

**Comment by fintzij** [comment]
Hi Simon,
There isn't a way to do this at the moment except by explicitly simulating and calculating the means and covariances of the LNA paths post-hoc. I've been meaning to refactor the code for a while and it would be good to have the option to return the LNA moments. I'll leave the issue open and try to implement this though.
Cheers,
jon

**Comment by sdwfrost** [comment]
Thanks Jon! I needed this for my toy implementation, but I think I've debugged what is going on with my math/code. I was trying to understand the noise term Φ^1/2 in equation (9) in your paper that gets fed into equation 13. In the Rcpp code, it's

```c++
diffusion.diag() += exp_neg_2Z % hazards;
```

and I couldn't quite reconcile it with equation (9). I had something like this:

```julia
# rates
λ = [β*S*I,γ*I]
# matrix with rates on the diagonal; elementwise square root (=matrix square root for diagonal matrices)
Λ = sqrt.(Diagonal(λ)) 
# Φ ^1/2 in equation 9
Φ = Diagonal(-exp.(Ñ))*Λ 
# equation 13
DΣ = D.(Σ) .~ F*Σ .+ Σ*F' .+ Φ*Φ' 
```

Plugging in a `Diagonal(exp.(-2*Ñ).* λ)` instead of `Φ*Φ'` in the above seems to work, but it would be great to check.

**Comment by fintzij** [comment]
Yup, this looks correct to me. Both of the matrices that multiply to get the SDE diffusion in eqn (9) are diagonal so making the substitution you suggest at the end of your post is correct and saves a bit on the matrix multiplication
